src/components/Buttons/ButtonLink.tsx

```diff
@@ -1,14 +1,28 @@
-import React from "react"
+import Button, { type ButtonProps } from "@/components/Buttons/Button"
+import { BaseLink, type LinkProps } from "@/components/Link"
 
-import { BaseLink, LinkProps } from "../Link"
+import { type MatomoEventOptions, trackCustomEvent } from "@/lib/utils/matomo"
 
-import type { IProps as IButtonProps } from "./Button"
-import Button from "./Button"
+export type ButtonLinkProps = LinkProps &
+  Omit<ButtonProps, "toId" | "onClick"> & {
+    customEventOptions?: MatomoEventOptions
+  }
 
-export type ButtonLinkProps = LinkProps & Omit<IButtonProps, "toId">
-
-const ButtonLink: React.FC<ButtonLinkProps> = (props) => {
-  return <Button as={BaseLink} activeStyle={{}} {...props} />
+const ButtonLink = ({ customEventOptions, ...props }: ButtonLinkProps) => {
+  const matomoEvent = customEventOptions ?? {
+    eventCategory: "Link",
+    eventAction: "clicked",
+    eventName: "ButtonLink clicked",
+    eventValue: (props.to ?? props.href) as string,
+  }
```

Comment:
Why are we adding this? do we want to keep track of every `ButtonLink` click? I'd lean towards removing this or discuss it with the team before we implement it.

Also, this is out of scope, we are changing behavior of the component.

Reply:
Notion card: [Add customEventOptions prop back to ButtonLink & Link](https://www.notion.so/efdn/Add-customEventOptions-prop-back-to-ButtonLink-Link-ef2e0044450b4399b681fce5b3ce937d?pvs=4)

Was related to this card

Reply:
This does add a default though; happy to remove this and just have an optional prop that can be passed.

Reply:
Got it, thanks.

We shouldn't need this in `ButtonLink`, as this action should be handled by the underlying `Link` component. The Link component already handles the matomo events, right? that is how we were doing it in the main repo.